st2common/st2common/router.py

```diff
@@ -513,6 +514,10 @@ def __init__(self, **entries):
 
         try:
             resp = func(**kw)
+        except StackStormDBObjectNotFoundError as e:
```

Comment:
Looks like with this change any DB query with DBObjectNotFound will result in a Warning now instead of more actionable log. It doesn't look like desired outcome to me as trying to fix one not found K/V issue we may introduce a regression.

We probably need only a warning when Key is not found in the datastore?

Reply:
Is a solution to this to add a new exception and have that raised @armab ?

Reply:
That would be a good idea to explore. For example, I see `TokenNotFoundError`, `ApiKeyNotFoundError` and similar examples in this file and that makes more sense.

Reply:
I think the solution is to really add a new exception such as: `DataStoreKeyNotFoundError`. The two you suggest aren't the precise meaning of them.

Reply:
`
class StackStormDBDataStoreKeyNotFoundError(StackStormBaseException):
    pass
`
Maybe would be a good exception in https://github.com/StackStorm/st2/blob/master/st2common/st2common/exceptions/db.py#L21?

Reply:
I was actually thinking that if you got the DBObjectNotFoundError while looking for a key, you would then raise a DataStoreKeyNotFoundError

Reply:
> The two you suggest aren't the precise meaning of them.

I hope that would be a good analogy and pointer where to look at and follow similar approach.

Reply:
I think this might be resolved.